src/redux/books/books.js

```diff
@@ -0,0 +1,23 @@
+import { createReducer, createAction } from '@reduxjs/toolkit';
+
+// Define action type for addding & removing
+export const ADD_BOOK = createAction('books/add');
+export const REMOVE_BOOK = createAction('books/remove');
+
+// set intial state to an empty array of books
+const initialState = {
+  books: [],
+};
+
+// Reducer
+const booksReducer = createReducer(initialState, (builder) => {
+  builder.addCase(ADD_BOOK, (state, action) => {
+    state.books.push(action.payload);
```

Comment:
- [ ] I will advise you to use a spread operator to create a new version of the array and then include the new book. This can be done by `return [...state.books, action.payload]`. With this, just a new version of the array would be returned and the state would not be tampered with.  Modifying the state directly will disturb this process.

React compares the previous state with the updated state to decide if the component needs to be re-rendered. Modifying the state directly will disturb this process. As a result the component will behave unexpectedly.

Reply:
Thank you for your good explanation - I have implemented your suggestions.